### DISCUSSION TITLE
Record animation in real time

### DISCUSSION THREAD

**Discussion by JekSun97** [godotengine/godot-proposals#9202] [Editor]
After XRBody support has now appeared in Godot (https://github.com/godotengine/godot/pull/88798), I think that it is necessary to add a recording button to the Animation window, we have a video recording button, now it’s time for animation!

This will allow you to animate humanoid characters directly in the engine using XR.

The principle is this:
select a model with a skeleton, press the record button in the Animation window, make movements, press the record button, clean the animation, and you're done!

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/234.

**Comment by goatchurchprime** [comment]
This can be generalized to puppetry where you are not limited to controlling only humanoid characters with humanoid inputs, but you can grab and move different parts of the rig with both hands.

I cannot believe that a system using the two 6DoF controllers to input the live animations of the rig cannot be many times more efficient and expressive than the current single 2D mouse motions keyframe procedure.

**Comment by ZenRepublic** [comment]
ouuu is that not possible yet?
